lib/BuildSystem/BuildSystem.cpp

```diff
@@ -772,6 +773,28 @@ class ProducedNodeTask : public Task {
     if (value.isMissingInput())
       return false;
 
+    // Check that the producer is currently recorded as a dependency
+    // This is an N x M scan, but we expect N (number of producers) to be 1
+    for (auto producer : node.getProducers()) {
```

Comment:
Would it make sense to instead use a "signature" like approach, where we take a signature of the producer and record that? OTOH, that would make the result size bigger, so its not obviously a win. I'm not thrilled about exposing the full result to each callback, but I also don't have any good argument against it.

I also think that this logic can be wrong if a node starts having multiple producers, maybe?

Reply:
If a node does start having multiple producers this should still be conservative and classify it as invalid if any of its producers is not currently listed as a dependency.